cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -402,22 +398,34 @@ public PackageSearchResult searchPackage(String query, String ballerinaVersion)
     }
 
     /**
-     * Create http URL connection.
+     * Create http URL connection and set required headers.
      *
-     * @param paths resource paths
+     * @param paths             resource paths
+     * @param supportedPlatform supported platform
+     * @param ballerinaVersion  ballerina version
      * @return http URL connection
      */
-    protected HttpURLConnection createHttpUrlConnection(String paths) throws ConnectionErrorException {
+    protected HttpURLConnection createHttpUrlConnection(String paths, String supportedPlatform, String ballerinaVersion)
+            throws ConnectionErrorException {
         URL url = convertToUrl(this.baseUrl + "/" + paths);
         try {
             // set proxy if exists.
             if (this.proxy == null) {
-                return (HttpURLConnection) url.openConnection();
+                return setRequiredHeaders((HttpURLConnection) url.openConnection(), supportedPlatform,
+                                          ballerinaVersion);
             } else {
-                return (HttpURLConnection) url.openConnection(this.proxy);
+                return setRequiredHeaders((HttpURLConnection) url.openConnection(this.proxy), supportedPlatform,
+                                          ballerinaVersion);
             }
         } catch (IOException e) {
             throw new ConnectionErrorException("Creating connection to '" + url + "' failed:" + e.getMessage());
         }
     }
+
+    private HttpURLConnection setRequiredHeaders(HttpURLConnection conn, String supportedPlatform,
+            String ballerinaVersion) {
+        conn.setRequestProperty(BALLERINA_PLATFORM, supportedPlatform);
+        conn.setRequestProperty(USER_AGENT, ballerinaVersion);
```

Comment:
We can directly use `RepoUtils.getBallerinaVersion()` for the `User-Agent` header instead of passing the value. Please check if its correct.

Reply:
For `pullPackage` in `CentralApiClient` we use the ballerina version for identifying whether it is a snapshot version or not, So we need to pass this for that. Thats why I kept all resources to pass the ballerina version.
If we need we can refactor other resources as you have mentioned in the above comment.

Reply:
Okay, I am okay with that.